library/src/main/scala/scala/scalajs/runtime/RuntimeLong.scala

```diff
@@ -302,9 +302,212 @@ final class RuntimeLong(val lo: Int, val hi: Int)
+     *   = 2^32*(ahi*blo / 2^32) + (ahi*blo % 2^32)
+     *   = 2^32*(ahi*blo / 2^32) + (ahi *[32] blo)
+     *
+     * Similarly,
+     *
+     * bhi*alo = 2^32*(alo*bhi / 2^32) + (alo *[32] bhi)
+     *
+     * Taking back our complete computation:
+     *
+     * a *[64] b
+     *   = ( 2^32*(ahi*blo + bhi*alo) + alo*blo ) % 2^64
+     *   = ( 2^64*(ahi*blo / 2^32) + 2^32*(ahi *[32] blo)
+     *        + 2^64*(alo*bhi / 2^32) + 2^32*(alo *[32] bhi)
+     *        + alo*blo ) % 2^64
+     *
+     * where distributing % 2^64 allows to get rid of the most awful parts:
+     *
+     *   = ( 2^32*(ahi *[32] blo) + 2^32*(alo *[32] bhi) + alo*blo) % 2^64
+     *
+     * Now we focus on the `alo*blo` part. We decompose it in 16-bit components.
+     *
+     * alo * blo
+     *   = 2^32*a1*b1 + 2^16*a1*b0 + 2^16*a0*b1 + a0*b0
+     *
+     * Because a1, a0, b1 and b0 are all <= 2^16-1, their pair-wise products
+     * are all <= (2^16-1)^2 = 2^32 - 2*2^16 + 1 < 2^32. This means that, for
+     * example,
+     *   a1*b0 = (a1*b0) % 2^32 = a1 *[32] b0
+     * with the same applying to other subproducts.
+     *
+     * Let
+     *   a1b1 = a1 *[32] b1
+     *   a1b0 = a1 *[32] b0
+     *   a0b1 = a0 *[32] b1
+     *   a0b0 = a0 *[32] b0
+     *
+     * We now have:
+     *
+     * alo * blo
+     *   = 2^32*a1b1 + 2^16*a1b0 + 2^16*a0b1 + a0b0
+     *
+     * We can decompose it using / and % as follows:
+     * alo * blo
+     *   = 2^32*((alo * blo) / 2^32) + ((alo * blo) % 2^32)
+     *
+     * Let
+     *   aloblo = (alo * blo) % 2^32
+     *   carry_from_lo_* = (alo * blo) / 2^32
+     *
+     * Then
+     * alo * blo = 2^32 * carry_from_lo_* + aloblo
+     *
+     * aloblo = (alo * blo) % 2^32
+     *   = (2^32*a1b1 + 2^16*a1b0 + 2^16*a0b1 + a0b0) % 2^32
+     *   = (2^16*a1b0 + 2^16*a0b1 + a0b0) % 2^32
+     *   = (((2^16*a1b0 % 2^32 + 2^16*a0b1 % 2^32) % 2^32) + a0b0 % 2^32) % 2^32
+     *   = (2^16*a1b0 % 2^32) +[32] (2^16*a0b1 % 2^32) +[32] (a0b0 % 2^32)
+     *   = (a1b0 << 16) +[32] (a0b1 << 16) +[32] a0b0
+     *
+     * carry_from_lo_* is more difficult.
+     *
+     * carry_from_lo_* = (alo * blo) / 2^32
+     *   = (2^32*a1b1 + 2^16*a1b0 + 2^16*a0b1 + a0b0) / 2^32
+     *   = a1b1 + (2^16*a1b0 + 2^16*a0b1 + a0b0) / 2^32
+     *   = a1b1 + (2^16*a1b0 + 2^16*a0b1 + (2^16*(a0b0 / 2^16) + (a0b0 % 2^16))) / 2^32
+     *   = a1b1 + (2^16*(a1b0 + a0b1 + (a0b0 / 2^16)) + (a0 % 2^16)) / 2^32
+     *             ----------------------------------
+     *                  multiple of 2^16
+     *   = a1b1 + ( (2^16*(a1b0 + a0b1 + (a0b0 / 2^16))) / 2^16 + (a0 % 2^16) / 2^16 ) / 2^16
+     *                                                             ---------
+     *                                                               < 2^16
+     *   = a1b1 + (a1b0 + a0b1 + (a0b0 / 2^16)) / 2^16
+     *   = a1b1 + (a1b0 + (a0b1 + (a0b0 >>> 16))) / 2^16
+     *                     ----    -----------
+     *                <= fffe0001    <= ffff
```

Comment:
Can you please:
- prefix with `0x` for better clarity.
- proof that `a0b1 <= 0xfffe0001` (enough to enhance line 368 IIUC).